### PR TITLE
Remove python constraints

### PR DESCRIPTION
**Pull request by YarShev**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by YarShev** [comment]
@h-vetinari, hi, could I get your help here please? Let me explain the problem. Prior to 0.5.0 unidist was built with `noarch`. There hasn't been any issue with builds. As of 0.5.0 unidist contains some cython extension(s). I forgot to remove the `noarch` step in #14 and new builds of unidist 0.5.0 were broken since python couldn't find those cython extentions. The error looks like.

```bash
ModuleNotFoundError: No module named 'unidist.core.backends.mpi.core._memory'
```

Then, I removed noarch in #15 and #16 and added some python constraints to not have builds with python 3.12 since Ray doesn't have it. Unfortunately, I got only three builds of [unidist 0.5.0](https://anaconda.org/conda-forge/unidist-mpi/files) package for python 3.11. In this PR I am trying to eventually get unidist 0.5.0 working. Could you take a look at the meta.yml? What am I missing to make unidist built for different python versions and different platforms.

**Comment by YarShev** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,12 +19,12 @@ outputs:
   - name: unidist-all
     requirements:
       host:
-        - python >=3.7.1,<3.12
+        - python >=3.7.1
```
Comment (anmyachev) [inline recipe/meta.yaml]: @YarShev  Looks like you need to remove any pins for python in `host` and `run` section.

Looks pretty much the same: https://github.com/bioconda/bioconda-recipes/pull/30938

One more link: https://github.com/bioconda/bioconda-recipes/issues/30693#issuecomment-939445165

Comment (YarShev) [reply recipe/meta.yaml]: Let me try.

**Comment by YarShev** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (YarShev) [reply recipe/meta.yaml]: Builds for a range of py versions were ran. That is a really frustrating issue. Thank you for figuring this out.

**Comment by h-vetinari** [comment]
> @h-vetinari, hi, could I get your help here please?

Seems you figured things out? In general, if you want to build per python version, don't specify a lower bound, but leave it as plain `- python`, like that conda smithy will correctly generate separate jobs

**Comment by YarShev** [comment]
@h-vetinari, yes, we figured out the root cause. I don't know if it is really an issue with conda or not, but it looks frustrating. Maintainers of other conda-forge feedstocks may see this issue in their feedstock at some point. It would be great to fix the issue or document it somewhere. Thanks for your response, btw.

**Pull request merged**